src/systeminfo_watcher.rs

```diff
@@ -0,0 +1,1130 @@
+use crate::{
+    self as fuel_telemetry, daemonise, enforce_singleton, info, into_recoverable, span,
+    telemetry_config, telemetry_formatter, EnvSetting, Level, Result, TelemetryError,
+    WatcherResult,
+};
+
+use nix::{
+    fcntl::{Flock, FlockArg},
+    sys::{
+        signal::{kill, Signal::SIGKILL},
+        stat::fstat,
+    },
+    time::ClockId,
+    unistd::{getpid, Pid},
+};
+use std::{
+    env::{set_var, var},
+    fs::{File, OpenOptions},
+    os::fd::{AsRawFd, RawFd},
+    path::{Path, PathBuf},
+    process::exit,
+    sync::{
+        atomic::{AtomicBool, AtomicI32, Ordering},
+        LazyLock,
+    },
+    time::Duration,
+};
+use sysinfo::{MemoryRefreshKind, System};
+use tracing_subscriber::{prelude::__tracing_subscriber_SubscriberExt, util::SubscriberInitExt};
+
+#[derive(Debug, Clone)]
+struct SystemInfoWatcherConfig {
+    // The path to its lockfile
+    lockfile: PathBuf,
+    // The path to its logfile
+    logfile: PathBuf,
+    // The timeout for the cloud provider metadata request
+    metadata_timeout: u64,
+    // The path to its touchfile
+    touchfile: PathBuf,
+    // The number of seconds to wait between collecting metrics
+    interval: u64,
+}
+
+fn config() -> Result<&'static SystemInfoWatcherConfig> {
+    static SYSTEMINFO_WATCHER_CONFIG: LazyLock<Result<SystemInfoWatcherConfig>> =
+        LazyLock::new(|| {
+            let get_env = |key, default| EnvSetting::new(key, default).get();
+
+            Ok(SystemInfoWatcherConfig {
+                // 60*60*24*30 = 2592000 (30 days)
+                interval: get_env("SYSTEMINFO_WATCHER_INTERVAL", "2592000").parse()?,
+                metadata_timeout: get_env("METADATA_TIMEOUT", "3").parse()?,
+                lockfile: Path::new(&telemetry_config()?.fuelup_tmp)
+                    .join("telemetry-systeminfo-watcher.lock"),
+                logfile: Path::new(&telemetry_config()?.fuelup_log)
+                    .join("telemetry-systeminfo-watcher.log"),
+                touchfile: Path::new(&telemetry_config()?.fuelup_tmp)
+                    .join("telemetry-systeminfo-watcher.touch"),
+            })
+        });
+
+    SYSTEMINFO_WATCHER_CONFIG
+        .as_ref()
+        .map_err(|e| TelemetryError::InvalidConfig(e.to_string()))
+}
+
+// Although there's no need to keep the state of this watcher, keep it consistent
+// with the other watchers
+pub struct SystemInfoWatcher;
+
+// Prevent recursive calls to start()
+static STARTED: AtomicBool = AtomicBool::new(false);
+
+/// The PID of the currently running `SystemInfoWatcher` daemon
+pub static PID: AtomicI32 = AtomicI32::new(0);
```

Comment:
nit: PIDs cannot be negative; use unsigned values instead:
```rust
pub static PID: AtomicU32 = AtomicU32::new(0);
```

Reply:
Although process PIDs are positive, the way how Unix signals process Groups is via negative PIDs